src/spherical_mesh.cc

```diff
@@ -52,12 +39,17 @@ double SphericalMesh::volume(int i) const
 
 double SphericalMesh::gradient(int idx, double f_lo, double f_hi) const
     {
-    if (idx == 0 && origin_ == 0.){
+    if (idx == 0 && lower_ == 0.){
```

Comment:
I think we can do this based on the BCs now. Actually, this is true for ALL of the classes, so move it to base Mesh. It also is true of the upper BC so it should be
```suggestion
    if ((idx == 0 && lower_bc_ == BoundaryType::reflect) ||
        (idx == shape_-1 && upper_bc_ == BoundaryType::reflect)
        {
```

So, move this to `Mesh` please!